### PR TITLE
Webpack improvements

### PR DESCRIPTION
**Pull request by ole1986**
webpack is also used for the extension (server)

While debugging webpack is instructed to bundle both server/browser into `dist` directory.

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -15,9 +15,22 @@ module.exports = {
         path: outPath,
         filename: 'bundle.js',
     },
-    performance: {
-        maxEntrypointSize: 716800,
-        maxAssetSize: 716800
+    optimization: {
+        namedChunks: true,
```
Comment (DonJayamanne) [inline webpack.config.js]: Why do we need to split?
Does it make a difference?

Comment (ole1986) [reply webpack.config.js]: I thought the whole browser page may appear quicker on slow network connections.
But I haven't really test this properly. Need to investigate more

Comment (DonJayamanne) [reply webpack.config.js]: I don't it'll make any difference, it's very small. But doesn't matter

Comment (ole1986) [reply webpack.config.js]: Ok. I actually do not like to have it split into chunks. So I am going to remove it later if you also think it does not make any differences

Thank you

File: src/extension.ts [hunk 2]
```diff
@@ -1,6 +1,4 @@
-if ((Reflect as any).metadata === undefined) {
-    require('reflect-metadata');
-}
+import 'reflect-metadata';
```
Comment (DonJayamanne) [inline src/extension.ts]:
```suggestion
// This line should always be right on top.
// Other extensions use this, re-importing will cause data stored by this to wipe out data in other extensions using this same package.
if ((Reflect as any).metadata === undefined) {
    require('reflect-metadata');
}
```

Comment (DonJayamanne) [reply src/extension.ts]: Doing that will break other extensions and possibly our extension as well.

**Comment by ole1986** [comment]
Any other ideas of a problemMatcher for webpack when running debug?

```
"problemMatcher": [
                {
                    "pattern": [
                        {
                            "regexp": ".",
                            "file": 1,
                            "location": 2,
                            "message": 3
                        }
                    ],
                    "background": {
                        "activeOnStart": true,
                        "beginsPattern": {
                            "regexp": "webpack is watching the files"
                        },
                        "endsPattern": {
                            "regexp": "Entrypoint main = extension.js"
                        }
                    }
                }
            ]
```

**Pull request merged**